code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx

```diff
@@ -352,12 +352,8 @@ export class PreviewWithSelection<TRenderer extends Renderer> extends Preview<TR
     try {
       await render.prepare();
     } catch (err) {
-      if (err !== PREPARE_ABORTED) {
-        // We are about to render an error so make sure the previous story is
-        // no longer rendered.
-        if (lastRender) await this.teardownRender(lastRender);
```

Comment:
I'm really unsure why the logic wouldn't have applied in the aborted cased too (the `lastRender` is out of date no matter what!). The tests from the PR that added this code https://github.com/storybookjs/storybook/pull/17599 still pass so I think it's good.